src/arc.rs

```diff
@@ -149,6 +149,40 @@ impl<T> Arc<T> {
         mem::forget(a);
         unsafe { Arc::from_raw(ptr) }
     }
+
+    /// Returns the inner value, if the [`Arc`] has exactly one strong reference.
+    ///
+    /// Otherwise, an [`Err`] is returned with the same [`Arc`] that was
+    /// passed in.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use triomphe::Arc;
+    ///
+    /// let x = Arc::new(3);
+    /// assert_eq!(Arc::try_unwrap(x), Ok(3));
+    ///
+    /// let x = Arc::new(4);
+    /// let _y = Arc::clone(&x);
+    /// assert_eq!(*Arc::try_unwrap(x).unwrap_err(), 4);
+    /// ```
+    pub fn try_unwrap(this: Self) -> Result<T, Self> {
+        if this
+            .inner()
+            .count
+            .compare_exchange(1, 0, Acquire, Relaxed)
```

Comment:
This should go via try_unique I think

Reply:
I think it's better here since the actual implementation of turning an arc into the inner data is something to do with arc's inner workings, plus the fact that `UniqueArc` is a wrapper/refinement of `Arc`. `UniqueArc::into_inner()` uses `Arc::try_unwrap()` and allows `Arc` to do it's internal <things>

Reply:
I guess? I dislike having this check in more places than it needs to be basically.

Reply:
Like, this impl can still be moved to UniqueArc and with a debug assertion and then we just use try_unique

Reply:
Done!